### ISSUE TITLE
Code quality: make contributions and featured activity use same code

### ISSUE THREAD

**Issue by neslihanturan** [commons-app/apps-android-commons#1566]
**Summary:** 

Currently Contributions Activity and Featured Activity includes several copy pasted codes. They should be organized by inheritance 

**Would you like to work on the issue?**

In two weeks, yes

**Comment by maskaravivek** [comment]
Yes, ideally contributions activity and featured activity could share a lot of code. Just the data source would be different, other than that the UI rendering, scrolling logic and list adapter logic can be shared. 

It would be great to pick up this cleanup as more and more browsing features are being added.

**Comment by misaochan** [comment]
Agreed, this would be useful.

**Comment by chrisdebian** [comment]
Have we got anybody currently contributing to the project, that is interested in code quality, and could potentially look at (refactor?) this?

Chris.
